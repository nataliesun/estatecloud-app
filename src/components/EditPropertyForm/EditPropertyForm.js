import React, { Component } from 'react';
import { withRouter } from 'react-router-dom'
import PropertyContext from '../../contexts/PropertyContext';
import PropertyApiService from '../../services/property-api-service';

class EditPropertyForm extends Component {
  static contextType = PropertyContext

  state = {
    currentProp: {}
  }

  static defaultProps = {
    match: {
      params: {}
    }
  }

  componentDidMount() {

    PropertyApiService.getProperty(this.props.match.params.property_id)
      .then(res => this.setState({
        currentProp: res
      }))


  }

  handleAddressChange = (value) => {
    this.setState({
      currentProp: {
        ...this.state.currentProp,
        address: value
      }
    })
  }

  handleSubmit = (ev) => {
    ev.preventDefault();

    // console.log(this.state.currentProp)

    PropertyApiService.editProperty(this.props.match.params.property_id, this.state.currentProp)
      .then(res => this.props.history.push('/dashboard'))
      .catch(this.context.setError)
  }

  handleStatusChange = (ev) => {
    this.setState({
      currentProp: {
        ...this.state.currentProp,
        status: ev.target.value
      }
    })
  }

  render() {
    return (
      <form className="EditPropertyForm" onSubmit={this.handleSubmit}>
        <div className="input-section">
          <label htmlFor="address">Address: </label>
          <input type="text" name="address" value={this.state.currentProp.address} onChange={e => this.handleAddressChange(e.target.value)} />
          <label htmlFor="city">City: </label>
          <input type="text" name="city" value={this.state.currentProp.city} onChange={e => this.handleCityChange(e.target.value)} />
          <label htmlFor="state">State: </label>
          <input type="text" name="state" value={this.state.currentProp.state} onChange={e => this.handleStateChange(e.target.value)} />
        </div>


        <div className="input-section">
          <label htmlFor="available">Available</label>
          <input
            type="radio"
            value="available"
            id="available"
            name="status"
            checked={this.state.currentProp.status === 'available'}
            onChange={this.handleStatusChange}
          />
          <label htmlFor="rented">Rented</label>
          <input
            type="radio"
            value="rented"
            id="rented"
            name="status"
            checked={this.state.currentProp.status === 'rented'}
            onChange={this.handleStatusChange}
          />
          <label htmlFor="occupied">Occupied</label>
          <input
            type="radio"
            value="occupied"
            id="occupied"
            name="status"
            checked={this.state.currentProp.status === 'occupied'}
            onChange={this.handleStatusChange}
          />
        </div>

        <div className="date-range">

        </div>

        <div className="input-section">
          <label htmlFor="rent-price">Rent price:</label>
          <input type="number" name="rent_price" value={this.state.rent_price} />


          <label htmlFor="intial-price">Initial price:</label>
          <input type="number" name="initial_price" />


          <label htmlFor="mortgage-payment">Mortgage payment: </label>
          <input type="number" name="mortgage_payment" placeholder="1500" />
        </div>

        <div className="submit-buttons">

          <button type="submit">Submit</button>
          <button type="reset">Reset</button>
        </div>
      </form >
    );
  }
}

export default withRouter(EditPropertyForm);