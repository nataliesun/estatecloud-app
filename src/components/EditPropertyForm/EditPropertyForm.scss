@import '../variables.scss';

.EditPropertyForm {
  padding: 0 20px;

  .input-section{
    display: flex;
    flex-direction: column;
    margin-bottom: 20px;

    label {
      margin-bottom: 10px;
      margin-top: 20px;
    }

    input[type=text], input[type=number] {
      @include big-input;
      
      &::placeholder {
        color: rgb(133, 133, 133);
      }
    }
  }

  .input-section-row {
    display: flex;
    
    input[type=radio] {
      margin-right: 20px;
      margin-left: 10px;
      margin-top: 3px;
    }
  }

  .submit-buttons button {
    @include highlighted-button;
    font-size: 16px;
    border: none;
    transition: .5s;

    &:hover {
      box-shadow: none;
      background: white;
      color: $button-purple;
      transform: translate(4px, 2px);
    }

    &:first-of-type {
      margin-right: 10px;
    }
  }
}