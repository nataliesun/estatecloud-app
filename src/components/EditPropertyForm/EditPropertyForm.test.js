import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import EditPropertyForm from './EditPropertyForm';

describe(`EditPropertyForm Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <EditPropertyForm />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <EditPropertyForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})