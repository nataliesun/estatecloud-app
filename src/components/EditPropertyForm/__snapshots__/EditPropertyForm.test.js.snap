// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPropertyForm Component renders the UI as expected 1`] = `
<form
  className="EditPropertyForm"
  onSubmit={[Function]}
>
  <div
    className="input-section"
  >
    <label
      htmlFor="address"
    >
      Address: 
    </label>
    <input
      name="address"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="city"
    >
      City: 
    </label>
    <input
      name="city"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="state"
    >
      State: 
    </label>
    <input
      name="state"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="input-section-row"
  >
    <label
      htmlFor="available"
    >
      Available
    </label>
    <input
      checked={false}
      id="available"
      name="status"
      onChange={[Function]}
      type="radio"
      value="available"
    />
    <label
      htmlFor="rented"
    >
      Rented
    </label>
    <input
      checked={false}
      id="rented"
      name="status"
      onChange={[Function]}
      type="radio"
      value="rented"
    />
    <label
      htmlFor="occupied"
    >
      Occupied
    </label>
    <input
      checked={false}
      id="occupied"
      name="status"
      onChange={[Function]}
      type="radio"
      value="occupied"
    />
  </div>
  <div
    className="date-range"
  />
  <div
    className="input-section"
  >
    <label
      htmlFor="rent-price"
    >
      Rent price:
    </label>
    <input
      name="rent_price"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <label
      htmlFor="intial-price"
    >
      Initial price:
    </label>
    <input
      name="initial_price"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <label
      htmlFor="mortgage-payment"
    >
      Mortgage payment: 
    </label>
    <input
      name="mortgage_payment"
      onChange={[Function]}
      placeholder="1500"
      type="number"
      value={0}
    />
  </div>
  <div
    className="submit-buttons"
  >
    <button
      type="submit"
    >
      Submit
    </button>
    <a
      href="/dashboard"
      onClick={[Function]}
    >
      <button>
        Cancel
      </button>
    </a>
  </div>
</form>
`;
